### PR TITLE
Consf 43 Serializing to XML

### PR DESCRIPTION
**Pull request by lizlux**
This PR adds serialization from our internal representation (IR) to XML. It also adds two new node types: caption and alt, which are meant to be included inside the image tags. 

Unit tests will be handled with https://wikia-inc.atlassian.net/browse/CONSF-58.

### CODE REVIEW

**Comment by BladeBronson** [comment]
I imagine this is right. @ChangeTip, give @lizlux kudos for the speed hacking.

**Comment by changetip** [comment]
Hi @lizlux, @BladeBronson sent you a tip worth 1 speed hacking ($0.50), and I'm here to deliver it ➔ **[collect your tip](https://www.changetip.com/collect/903815)**.

1 speed hacking

**[Learn more about ChangeTip](https://www.changetip.com/tip-online/github)**

**Comment by lizlux** [comment]
@BladeBronson lol thanks!

File: src/serializers/xml/helpers.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+'use strict';
+
+export function equals(leftVal, rightVal, options) {
+	if (arguments.length < 3)
+		throw new Error("Handlebars Helper equal needs 2 parameters");
+	if( leftVal != rightVal ) {
```
Comment (kenkouot) [inline src/serializers/xml/helpers.js]: `!==` and ws inside parens

File: src/core.js [hunk 2]
```diff
@@ -44,9 +44,13 @@ class Core extends Model {
 		}
 	}
 
+	getSerializedData() {
```
Comment (kenkouot) [inline src/core.js]: can you change this method name back to `serialize`, it's meant to be an abstract method.

File: src/types/alt.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+'use strict';
```
Comment (kenkouot) [inline src/types/alt.js]: Discussed offline, this is just going to be a property on the image class.

File: src/serializers/xml/serializer.js [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+	return template(data);
+}
+
+/**
+ * deserialize
+ *
+ * @param doc {string} A string of portable infobox xml
+ * @return {object} an object containing new instances of InfoboxData and InfoboxThemeData
+ */
+export function deserialize(doc) {
```
Comment (kenkouot) [inline src/serializers/xml/serializer.js:28]: Please revert the this test code.

File: src/validators.js [hunk 5]
```diff
@@ -5,3 +5,7 @@ export function isString(input) {
 export function isNumeric(input) {
 	return !isNaN(parseFloat(n)) && isFinite(n);
 }
+
+export function isObject(input) {
+	return typeof input === 'object';
```
Comment (kenkouot) [inline src/validators.js]: This isn't robust enough. `typeof [] === 'object' => true`. You probably want to use `Object.getPrototypeOf`

**Comment by kenkouot** [comment]
lgtm

**Pull request merged**